### PR TITLE
Add CRUD support for custom policy resource

### PR DESCRIPTION
**Pull request by Vasundharashukla**
1. **What this PR does / why we need it**:
Support functions of Create, Update, Read and Delete to provision custom policies.

3. **Additional information**

### CODE REVIEW

File: internal/resources/policy/type/custom/resource_custom_policy.go [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+func resourceCustomPolicyCreate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	customPolicyName, _ := d.Get(nameKey).(string)
+
+	scopedFullnameData := policy.ConstructScope(d, customPolicyName)
+
+	if scopedFullnameData == nil {
+		return diag.Errorf("Unable to create Tanzu Mission Control cluster custom policy entry; Scope full name is empty")
+	}
```
Comment (ishangupta-ds) [inline internal/resources/policy/type/custom/resource_custom_policy.go:65]: Please try to reuse these error formats across security and custom policies by declaring them at the policy level.

File: internal/resources/policy/type/custom/resource_custom_policy.go [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+		return diag.Errorf("no valid scope type block found: minimum one valid scope type block is required among: %v. Please check the schema.", strings.Join(policy.ScopesAllowed[:], `, `))
+	}
+
+	// always run
+	d.SetId(UID)
+
+	return resourceCustomPolicyRead(ctx, d, m)
+}
+
+func retrieveCustomPolicyUIDMetaAndSpecFromServer(config authctx.TanzuContext, scopedFullnameData *policy.ScopedFullname, d *schema.ResourceData, customPolicyName string) (string, *objectmetamodel.VmwareTanzuCoreV1alpha1ObjectMeta, *policymodel.VmwareTanzuManageV1alpha1CommonPolicySpec, error) {
```
Comment (ishangupta-ds) [inline internal/resources/policy/type/custom/resource_custom_policy.go:131]: Reuse this function across security and custom policies.

File: internal/resources/policy/type/custom/resource_custom_policy.go [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+	}
+
+	if err := d.Set(specKey, flattenSpec(spec)); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diags
+}
+
+func resourceCustomPolicyInPlaceUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
```
Comment (ishangupta-ds) [inline internal/resources/policy/type/custom/resource_custom_policy.go:272]: Reuse this function across security and custom policies.

**Review by ishangupta-ds — CHANGES_REQUESTED** [review]

**Review by ishangupta-ds — APPROVED** [review]
LGTM.

**Review by shreyassreenivas — APPROVED** [review]

**Pull request merged**